File: frontend/src/app/models/bill/Bill.ts
import { BaseEntity } from '../base/BaseEntity';
import { EOrderType } from '../order/EOrderType';
import { EOrderStatus } from '../order/EOrderStatus';
import { Order } from '../order/Order';

export class Bill extends BaseEntity {
  billType: EOrderType;
  billStatus: EOrderStatus;
  billTotalPrice: number;

  order: Order;

  constructor(
    billType: EOrderType,
    billStatus: EOrderStatus,
    billTotalPrice: number,
    order: Order
  ) {
    super();
    this.billType = billType;
    this.billStatus = billStatus;
    this.billTotalPrice = billTotalPrice;
    this.order = order;
  }
}

File: backend/src/main/java/com/springbootjwtpostgres/backend/bill/Bill.java
package com.springbootjwtpostgres.backend.bill;

import com.springbootjwtpostgres.backend.basemodels.BaseEntity;
import com.springbootjwtpostgres.backend.order.Order;
import com.springbootjwtpostgres.backend.order.OrderStatus;
import com.springbootjwtpostgres.backend.order.OrderType;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@Entity
@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class Bill extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private OrderType billType;
    private OrderStatus billStatus;
    private double billTotalPrice;
    private Date createdAt;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "order_id")
    private Order order;

    @Column(name = "order_id", insertable = false, updatable = false)
    private Long orderId;
}
